### PR TITLE
2023-07 python dependency updates

### PR DESCRIPTION
**Pull request by ialarmedalien**
# Description of PR purpose/changes

The usual python updates. Also added in isort, which sorts dependencies for the ultimate in pedantry and attention to detail.

Ran isort and black and added the resulting changes to this PR as the second commit. 

# Jira Ticket / Issue #

N/A

# Testing Instructions
* Details for how to test the PR:
- [x] Tests pass locally and in GitHub Actions
- [-] Changes available by spinning up a local narrative and navigating to _X_ to see _Y_

# Dev Checklist:

- [x] My code follows the guidelines at https://sites.google.com/lbl.gov/trussresources/home?authuser=0
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [-] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] (Python) I have run Black and Flake8 on changed Python code manually or with a git precommit hook

# Updating Version and Release Notes (if applicable)

- [-] [Version has been bumped](https://semver.org/) for each release
- [x] [Release notes](/RELEASE_NOTES.md) have been updated for each release (and during the merge of feature branches)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -118,8 +118,8 @@
         "defaults"
     ],
     "scripts": {
-        "black": "black -v --exclude '.+Impl.py' .",
```
Comment (ialarmedalien) [inline package.json:121]: there are no impl files in the narrative repo

File: src/biokbase/narrative/jobs/job_new.py
Comment (briehl) [inline src/biokbase/narrative/jobs/job_new.py:1]: What's this? Is this an accidental addition?

Comment (ialarmedalien) [reply src/biokbase/narrative/jobs/job_new.py:1]: yes. accidental addition. Well spotted!

**Review by briehl — APPROVED** [review]
Looks good.

**Pull request merged**